website/src/components/SearchPage/DownloadDialog/DownloadDialog.spec.tsx

```diff
@@ -143,6 +144,44 @@ describe('DownloadDialog', () => {
         expect(screen.getByLabelText(gzipCompressionLabel)).toBeInTheDocument();
     });
 
+    test('should show copy URL button when using GET request', async () => {
+        await renderDialog();
+        await checkAgreement();
+
+        const copyUrlButton = screen.getByTestId('copy-download-url');
+        expect(copyUrlButton).toBeInTheDocument();
+        expect(copyUrlButton).toHaveAttribute('title', 'Copy download URL');
+    });
+
+    test('should not show copy URL button when using POST request', async () => {
+        await renderDialog({
+            downloadParams: new SelectFilter(new Set<string>(['x'.repeat(approxMaxAcceptableUrlLength * 2)])),
+        });
+        await checkAgreement();
+
+        expect(screen.queryByTestId('copy-download-url')).not.toBeInTheDocument();
+    });
+
+    test('should copy the right URL when clicking on the copy button', async () => {
+        const clipboardMock = vi.spyOn(navigator.clipboard, 'writeText').mockResolvedValue();
+
+        await renderDialog();
+        await checkAgreement();
+
+        const copyUrlButton = screen.getByTestId('copy-download-url');
+        await userEvent.click(copyUrlButton);
+        expect(clipboardMock).toHaveBeenCalledTimes(1);
+        const copiedText = clipboardMock.mock.calls[0][0];
+        expect(
+            copiedText.startsWith(
+                'https://lapis/sample/details?downloadAsFile=true&downloadFileBasename=ebola_metadata_',
+            ),
+        ).toBe(true);
+        expect(
+            copiedText.endsWith('&versionStatus=LATEST_VERSION&isRevocation=false&dataUseTerms=OPEN&dataFormat=tsv'),
+        ).toBe(true);
```

Comment:
```suggestion
        ).toBe(true);
        
        clipboardMock.mockRestore();
```

To avoid any potential side effects.